added github workflow for building and pushing base images

engelmi opened the pull request:
This PR introduces a new workflow to build and push the BlueChi base image to quay.io. Its only triggered manually at the moment, but can be easily extended to be built in a certain interval (e.g. every day) to keep up with changes from selinux etc.

engelmi committed:
added github workflow for building and pushing base images

Signed-off-by: Michael Engel <mengel@redhat.com>

mwperina commented on .github/workflows/images.yml:
```diff
@@ -0,0 +1,30 @@
+
+      - name: Quay login
+        run: |
+          echo ${{ secrets.QUAY_BOT_API_TOKEN }} | buildah login -u="bluechi+bluechi_bot" --password-stdin quay.io
```
Wouldn't the API token be visible in the execution logs of the workflow?

ygalblum replied:
Maybe a better way to implement this is to split the build_and_push script so that it only builds the image and then use https://github.com/marketplace/actions/push-to-registry to push it

engelmi replied:
> Wouldn't the API token be visible in the execution logs of the workflow?

@mwperina Yes and no. Its GitHub Action specific, but they will mask any secrets. 

@ygalblum Good point. I was too lazy to refactor the script, to be honest. But lets use that action and not rely solely on gh masking secrets.

engelmi replied:
@mwperina @ygalblum Could you have another look?
I know there is also https://github.com/marketplace/actions/buildah-build for building the images/manifest, but wanted to still use the bash script.

engelmi committed:
refactored build and push script for base images

Split build and push into two distinct functions. Also, renamed
functions and added an parameter check.

Signed-off-by: Michael Engel <mengel@redhat.com>

mwperina commented on tests/README.md:
```diff
@@ -197,8 +197,8 @@ From the root directory of the project run the following commands:
 # In order to build and directly push, login first
 buildah login -u="someuser" -p="topsecret" quay.io
-PUSH_MANIFEST=yes bash build-scripts/build-push-containers.sh build_base
+PUSH_MANIFEST=yes bash build-scripts/build-push-containers.sh build-base
```
Do we need to run bash from bash?

```
# In order to build and directly push, login first
$ buildah login -u="someuser" -p="topsecret" quay.io
$ PUSH_MANIFEST="yes" ./build-scripts/build-push-containers.sh build-base

```

engelmi replied:
I'd need to make the script executable, but otherwise it works either way, I think. I don't have a preference here - should I change it?

mwperina replied:
Yes please, it's a shell script, so it should be executable, I don't a point why to run shell script using another instance of bash, when you are already in a bash instance

mkemel replied:
Just note that in this case the executed script will share the variables with the calling script, and executing `exit` will exit the calling script as well.

mkemel replied:
It's not a problem, just something to be conscious about

mkemel replied:
IIRC `./script.sh` also executes in a new process.
To execute in the same bash instance `source script.sh` or `. script.sh` should be used

engelmi replied:
Yes. I think only the scripts involved in building the rpm are used in the same instance. In all other locations the scripts are rather distinct from each other.

mkemel replied:
Yes, e.g. `build-rpm.sh` runs `build-srpm.sh` with `source` command.
What I meant is I don't think executing a script by `bash build-scripts/build-push-containers.sh` or by `./build-scripts/build-push-containers.sh` makes a difference bash-process-wise...

mkemel replied:
But I might be wrong

mkemel replied:
https://dev.to/amrhrabdeen/difference-between-executing-a-script-and-sourcing-it-in-linux-570f

mwperina replied:
It's not, but it looks nicer in my opinion

mkemel replied:
From that point of view - yes, most definitely :)

engelmi committed:
make all build-scripts executable

Signed-off-by: Michael Engel <mengel@redhat.com>

mwperina reviewed (approved):
LGTM
